sdlf-ddk-lightweight/data_lake/stages/sdlf_light_transform.py

```diff
@@ -0,0 +1,678 @@
+            self,
+            f"{self._prefix}-{team}-{pipeline}-error-a",
+            environment_id = self._environment_id,
+            function_name=f"{self._prefix}-{team}-{pipeline}-error-a",
+            code=Code.from_asset(os.path.join(f"{Path(__file__).parents[1]}", "src/lambdas/sdlf_light_transform/error")),
+            handler="handler.lambda_handler",
+            role = error_lambda_role,
+            description="send errors to DLQ",
+            timeout=cdk.Duration.minutes(10),
+            memory_size=256,
+            runtime = Runtime.PYTHON_3_9,
+            layers = [self._config.data_lake_lib]
+        )
+ 
+        error_task = LambdaInvoke(
+            self,
+            f"{self._prefix}-{team}-{pipeline}-error-task",
+            lambda_function=error_lambda,
+            result_path=None
+        )
+        return error_task
+
+    def _create_redrive_lambda(self, team, pipeline) -> None:
+        redrive_lambda_role = Role(
+            self,
+            f"{self._prefix}-redrive-role-{team}-{pipeline}-a",
+            role_name=f"{self._prefix}-redrive-role-{team}-{pipeline}-a",
+            assumed_by=ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[ManagedPolicy.from_aws_managed_policy_name("service-role/AWSLambdaBasicExecutionRole")],
+        )
+        ManagedPolicy(
+            self,
+            f"{self._prefix}-redrive-policy-{team}-{pipeline}-a",
+            managed_policy_name = f"{self._prefix}-redrive-policy-{team}-{pipeline}-a",
+            roles=[redrive_lambda_role],
+            document=PolicyDocument(
+                statements=[
+                PolicyStatement(
+                        effect=Effect.ALLOW,
+                        actions=[
+                            "ssm:GetParameter",
+                            "ssm:GetParameters"
+                        ],
+                        resources=[f"arn:aws:ssm:{cdk.Aws.REGION}:{cdk.Aws.ACCOUNT_ID}:parameter/*"],
+                    )
+                ]
+            ),
+        )
+        self._routing_dlq.grant_consume_messages(redrive_lambda_role)
+        self._routing_queue.grant_send_messages(redrive_lambda_role)
+
+        LambdaFactory.function(
+            self,
+            f"{self._prefix}-{team}-{pipeline}-redrive-a",
+            environment_id = self._environment_id,
+            function_name=f"{self._prefix}-{team}-{pipeline}-redrive-a",
+            code=Code.from_asset(os.path.join(f"{Path(__file__).parents[1]}", "src/lambdas/sdlf_light_transform/redrive")),
+            handler="handler.lambda_handler",
+            environment={
+                "TEAM": self.team,
+                "PIPELINE": self.pipeline,
+                "STAGE": "StageA"
+            },
+            role = redrive_lambda_role,
+            description="Redrive Step Function stageA",
+            timeout=cdk.Duration.minutes(10),
+            memory_size=256,
+            runtime = Runtime.PYTHON_3_9,
```

Comment:
I wonder if it makes sense to add the runtime version as a parameter for the stage. That way, when Python 3.9 becomes outdated, the customer only needs to change the runtime in one place (per transformation pipeline at least).